backend/src/main/java/gov/cdc/usds/simplereport/db/repository/TestEventRepository.java

```diff
@@ -21,4 +22,14 @@
 	// Need to control how this query is built. "between" is too vague.
 	@Query("FROM #{#entityName} q WHERE q.createdAt > :before AND q.createdAt <= :after ORDER BY q.createdAt DESC ")
 	public List<TestEvent> queryMatchAllBetweenDates(Date before, Date after);
+
+	@Query( "FROM #{#entityName} q "
+			+ "WHERE q.organization = :org "
+			+ "AND q.organization.isDeleted = false "
+			+ "AND q.patient.isDeleted = false "
```

Comment:
I think this is a bug. I think we want to show test results even if the patient has been archived

@aliciabeckett-gov?

Reply:
Excellent catch. Seems like a reasonable assumption to show them.

Reply:
Pulling @benwarfield-usds into this discussion.

 Starting query

```
@Query( "FROM #{#entityName} q "
			+ "WHERE q.organization = :org "
			+ "AND q.organization.isDeleted = false "
			+ "AND q.facility = :facility "
			+ "AND q.correctionStatus = 'ORIGINAL' "
			+ "ORDER BY q.createdAt ")
	public List<TestEvent> getTestEventResults(Organization org, Facility facility);
```

Another Q: Should matching on facility be enough?
Are we matching on both as a security thing?

I think the actual final query that matches the latest event for each order would look more like this:

```
SELECT *
FROM
    (SELECT MAX(created_at) each_created_at
        FROM simple_report.test_event inner_evt
        WHERE inner_evt.facility = :facility
     GROUP BY test_order_id) AS latest_events
INNER JOIN
  simple_report.test_event
ON
  simple_report.test_event.created_at = latest_events.each_created_at
ORDER BY created_at
```

Reply:
This works, but has a hard coded `entityName` _aka `#{#entityName}`_

notice the `nativeQuery = true`

```
	@Query( value = "SELECT * FROM " +
			"    (SELECT MAX(created_at) each_created_at " +
			"        FROM simple_report.test_event inner_evt " +
			"        WHERE inner_evt.facility_id = :facility_id " +
			"     GROUP BY test_order_id) AS latest_events " +
			"INNER JOIN simple_report.test_event events " +
			"ON events.created_at = latest_events.each_created_at " +
			"ORDER BY created_at", nativeQuery = true)
	public List<TestEvent> getTestEventResults(@Param("facility_id")UUID facility_id);
```